pkgs/tools/misc/tremor-rs/default.nix

```diff
@@ -27,8 +27,8 @@ rustPlatform.buildRustPackage rec {
 
   LIBCLANG_PATH = "${llvmPackages.libclang}/lib";
 
-  # OPENSSL_NO_VENDOR - If set, always find OpenSSL in the system, even if the vendored feature is enabled.
-  OPENSSL_NO_VENDOR = 1;
+  # OPENSSL_STATIC - If set, the crate will statically link to OpenSSL rather than dynamically link.
+  OPENSSL_STATIC = 1;
```

Comment:
I don't think static linking is desirable? Even with dynamic linking, the specific OpenSSL version is pinned with Nix, and static linking would only waste a bit of space. It appears to work fine without this flag.

Reply:
Thanks @stephank I'll remove that.
btw I'm confused about the Darwin terminologies (maybe because I'm not a darwin user) in nixpkgs and I'm confused what Security.framwork/CoreFoundation etc. Can you point me to any piece of documentation where I can read up on this stuff?

Reply:
- What I was showing, `otool -L ...`, is the equivalent of `ldd ...` on Linux. 
 - A `*.dylib` (dynamic library) file is the same as an `*.so` (shared object) on Linux.
 - A 'framework bundle' is a directory that can combine a dylib with other associated files, often the C header files, but anything really.
 - In general, a 'bundle' on Darwin is directory that is presented as a single file. Frameworks are just one kind.
 - libSystem is the C standard library, what would be glibc on Linux.
 - CoreFoundation is a library that provides a lot of low-level stuff the system depends on: data structures, XML parsing, URL handling, etc.
 - Security.framework is the equivalent of OpenSSL, implementing cryptographic algorithms and TLS.

Some links:
 - https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/FrameworkAnatomy.html
 - https://developer.apple.com/documentation/corefoundation
 - https://en.wikipedia.org/wiki/Bundle_(macOS)
 - https://en.wikipedia.org/wiki/Core_Foundation

The official docs can be a bit dense, though. Also not sure how useful this is without a macOS system to experiment with. 🙃

I have changes ready for Darwin support for this package, and can make a PR later. Just have to wait for this one to go through, because it conflicts.

Reply:
Can you change it back to not statically linking?

Reply:
It looks like those files are not currently installed anywhere. You probably want to extend `postInstall` to place those in `$out/lib/tremor/tremor-script/lib`.

On top of that, I think what we want is to change this default `TREMOR_PATH` to point to the final `/nix/store` path (`$out`)  instead of `/usr/lib`: https://github.com/tremor-rs/tremor-runtime/blob/24bb6a05b4c303c3d6bbb743a3f6d58c0e73a5fa/tremor-script/src/path.rs#L59

My first thought would be to add a `postPatch` that does this using `sed` (a regular patch won't work, because `$out` is dynamic), but I'm not sure if that's the best / correct way. Also not sure who can advise here. 🤔

Reply:
AFAIK the `/usr/lib` path is assumed only when no environment variable is set so that shouldn't be a problem.

Reply:
The only alternative approach I’ve seen is to make `tremor` a wrapper script that provides `TREMOR_PATH`.

Actually setting an environment variable in the user’s shell is in the domain of NixOS / nix-darwin / home-manager modules.

Out of those options, just patching the default in the code seems to me like the least cludgey / least surprising to users. It’s not unusual for Nix packages to do this, I believe.